src/validation/variable.rs

```diff
@@ -1,16 +1,123 @@
-use plc_ast::ast::{ArgumentProperty, Pou, PouType, Variable, VariableBlock, VariableBlockType};
+use plc_ast::ast::{
+    ArgumentProperty, ConfigVariable, Pou, PouType, Variable, VariableBlock, VariableBlockType,
+};
 use plc_diagnostics::diagnostics::Diagnostic;
 
 use super::{
     array::validate_array_assignment,
-    statement::{validate_enum_variant_assignment, validate_pointer_assignment, visit_statement},
+    statement::visit_statement,
     types::{data_type_is_fb_or_class_instance, visit_data_type_declaration},
     ValidationContext, Validator, Validators,
 };
 use crate::index::VariableIndexEntry;
 use crate::typesystem::DataTypeInformation;
+use crate::validation::statement::validate_enum_variant_assignment;
+use crate::validation::statement::validate_pointer_assignment;
 use crate::{index::const_expressions::ConstExpression, resolver::AnnotationMap};
 
+pub fn visit_config_variable<T: AnnotationMap>(
+    validator: &mut Validator,
+    var_config: &ConfigVariable,
+    context: &ValidationContext<T>,
+) {
+    if var_config.name_segments.is_empty() {
+        return;
+    }
+
+    let segments: Vec<&str> = var_config.name_segments.iter().map(|segment| segment.as_str()).collect();
+    let segments: &[&str] = &segments;
```

Comment:
Sadly (afaik) yes, the `find_variable` method expects a `&[&str]` here

Reply:
I think line 28 might be obsolete, since after transforming the vector simply borrowing the `Vec<&str>` will coerce to `&[&str]` when passing it as `&segments`, but I might be wrong. I can give it a quick try.